### PR TITLE
Upgrade RustPython

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This pulls in https://github.com/astral-sh/RustPython-Parser/pull/5 and https://github.com/astral-sh/RustPython-Parser/pull/7. It changes RustPython to include the `(` and `)` in the `arguments` range and introduces a new `Decorator` node.

## Test Plan

* A few comments got reordered in the snapshot tests because they are ordered by the node's start position.
* The range of some decorator messages now include the `@`. I think that's preferred. Let me know if you prefer the way it was before.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4898** <a href="https://app.graphite.dev/github/pr/charliermarsh/ruff/4898" target="_blank">Graphite</a> 
    * **PR #4900** <a href="https://app.graphite.dev/github/pr/charliermarsh/ruff/4900" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/charliermarsh/ruff/4900?utm_source=stack-comment).

**Review by charliermarsh — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [workspace]
 members = ["crates/*"]
+resolver = "2"
```
Comment (charliermarsh) [inline Cargo.toml:3]: Why is this necessary?

Comment (MichaReiser) [reply Cargo.toml:3]: Rust nightly complains because some workspace members use `edition 2021` but the workspace does not. This fixes this warning (yes, I'm now using nightly because I can't live without CLions test runner integration)
